src/generator/generator_visitor.ts

```diff
@@ -427,23 +427,51 @@ class GeneratorVisitor extends Visitor {
     private getArrayElement(node: MemberExpression) {
         super.visitMemberExpression(node);
 
-        // The offset can not be used, because the memberaccess value can also be a mathematical term or a variable
-        return this.module.i32.load(0, 4, this.getPointer());
+        const type = this.getExpressionType(node);
+
+        switch (type) {
+            case WebAssemblyType.INT_32:
+                return this.module.i32.load(0, 4, this.getPointer(4));
+            case WebAssemblyType.FLOAT_64:
+                return this.module.f64.load(0, 8, this.getPointer(8));
+            default:
+                throw new Error(`Unknown type of member expression: ${type}`);
```

Comment:
This only prints a number as the type in the error message. Use `WebAssemblyType[type]` instead.